integration/preemption.py

```diff
@@ -0,0 +1,113 @@
+"""Preemption tests, need to be run on a an actual cluster"""
+import logging
+import shutil
+import subprocess
+import time
+from datetime import datetime
+from pathlib import Path
+
+import submitit
+from submitit import AutoExecutor, Job
+from submitit.core import test_core
+
+FILE = Path(__file__)
+LOGS = FILE.parent.parent / "log_test" / f"{FILE.stem}_log"
+
+log = logging.getLogger("preemption_main")
+formatter = logging.Formatter("%(name)s %(levelname)s (%(asctime)s) - %(message)s")
+handler = logging.StreamHandler()
+handler.setFormatter(formatter)
+log.setLevel(logging.INFO)
+log.addHandler(handler)
+
+
+def clock(partition: str, duration: int):
+    log = logging.getLogger(f"preemption_{partition}")
+    tick_tack = ["tick", "tack"]
+    try:
+        for minute in range(duration - 5):
+            log.info(tick_tack[minute % 2])
+            time.sleep(60)
+        logging.warning("*** Exited peacefully ***")
+        return duration
+    except:
+        logging.warning(f"!!! Interrupted on: {datetime.now().isoformat()}")
+        raise
+
+
+def pascal_job(partition: str, timeout_min: int, node: str = "") -> Job:
+    """Submit a job with specific constraint that we can preempt deterministically."""
+    ex = submitit.AutoExecutor(folder=LOGS, slurm_max_num_timeout=1)
+    ex.update_parameters(
+        name=f"submitit_preemption_{partition}",
+        timeout_min=timeout_min,
+        mem_gb=7,
+        slurm_constraint="pascal",
+        slurm_comment="submitit integration test",
+        slurm_partition=partition,
+        # pascal nodes have 80 cpus.
+        # By requesting 50 we now that their can be only one such job with this property.
+        cpus_per_task=50,
+        slurm_additional_parameters={},
+    )
+    if node:
+        ex.update_parameters(slurm_additional_parameters={"nodelist": node})
+
+    return ex.submit(clock, partition, timeout_min)
+
+
+def wait_job_is_running(job: Job) -> None:
+    while job.state != "RUNNING":
```

Comment:
may be safer check if it is still PENDING or UNKNOWN (can't remember if that's exactly it). If it fails for some unknown reason, this could wait forever